src/filters/GeltungsbereicheMenuFilter/GeltungsbereicheMenuFilter.js

```diff
@@ -0,0 +1,174 @@
+import {
+  InputLabel,
+  MenuItem,
+  IconButton,
+  FormControl,
+  makeStyles,
+} from '@material-ui/core';
+import { MdClear } from 'react-icons/md';
+import Select from '../../components/Select';
+import { geltungsbereicheDataLayer } from '../../config/layers';
+
+const useStyles = makeStyles(() => ({
+  selectWrapper: {
+    position: 'relative',
+  },
+  formControl: {
+    padding: '20px 0',
+  },
+  paper: {
+    overflow: 'hidden auto',
+    maxHeight: 350,
+    width: 268,
+  },
+  item: {
+    display: 'block',
+    textOverflow: 'ellipsis',
+  },
+  input: {
+    width: 270,
+  },
+  value: {
+    maxWidth: 190,
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  clearBtn: {
+    position: 'absolute',
+    top: 20,
+    right: 30,
+  },
+}));
+
+const nullOption = {
+  value: 'gb-null',
+  label: 'Alle Produkte',
+};
+
+const GeltungsbereicheMenuFilter = ({ topic }) => {
+  const { t } = useTranslation();
+  const apiKey = useSelector((state) => state.app.apiKey);
+  const [filterValue, setFilterValue] = useState(nullOption);
+  const [productChoices, setProductChoices] = useState([]);
+  const classes = useStyles();
+
+  useEffect(() => {
+    fetch(
+      `${geltungsbereicheDataLayer.url}/data/ch.sbb.geltungsbereiche.json?key=${apiKey}`,
+    )
+      .then((res) => res.json())
+      .then((data) => {
+        const choices = [
+          nullOption,
+          ...Object.keys(data['geops.geltungsbereiche']).map((key) => {
+            return {
+              value: key,
+              label: data['geops.geltungsbereiche'][key],
+            };
+          }),
+        ];
+        setProductChoices(choices);
+      })
+      .catch((err) =>
+        // eslint-disable-next-line no-console
+        console.error(
+          err,
+          new Error('Failed to fetch ch.sbb.geltungsbereiche.json'),
+        ),
+      );
+  }, [apiKey]);
+
+  const layers = topic.layers.filter((l) => {
+    return /^ch.sbb.geltungsbereiche-/.test(l.key);
+  });
+
+  const onChange = (value) => {
+    layers.flat().forEach((layer) => {
+      const { mbMap } = layer.mapboxLayer;
+      const style = mbMap.getStyle();
+
+      if (!mbMap || !style) {
+        return;
+      }
+      const styleLayer = style.layers.find((l) => {
+        return layer.styleLayersFilter(l);
+      });
+      const newFilter = styleLayer.filter.filter((fil) => {
+        // Remove previous product filters.
+        if (
+          Array.isArray(fil) &&
+          Array.isArray(fil[2]) &&
+          fil[2][1] === 'geltungsbereiche_str'
+        ) {
+          return false;
+        }
+        return true;
+      });
+
+      if (value && value !== nullOption.value) {
+        // Add new filter
+        newFilter.push(['in', value, ['get', 'geltungsbereiche_str']]);
+      }
+
+      mbMap.setFilter(styleLayer.id, newFilter);
+    });
+
+    setFilterValue(productChoices.find((opt) => opt.value === value));
+  };
+
+  return productChoices.length ? (
```

Comment:
It's more understanadabel  if you separtare the null case
```suggestion
  if (!productChoices.length) {
    return null:
  }
  return (
```